src/trove/chart.js.rkt

```diff
@@ -0,0 +1,835 @@
+    (data-spec
+      (name "ChartWindow")
+      (type-vars ())
+      (variants ("bar-chart-window"))
+      (shared
+        ((method-spec
+          (name "title")
+          (arity 2)
+          (params ())
+          (contract (a-arrow ,Self ,S ,ChartWindow))
+          (args ("self" "tiltle"))
+          (return ,ChartWindow)
+          (doc ("Construct a new " ,ChartWindow " with a new title. "
+                "By default, the title will empty")))
+        (method-spec
+          (name "width")
+          (arity 2)
+          (params ())
+          (contract (a-arrow ,Self ,N ,ChartWindow))
+          (args ("self" "width"))
+          (return ,ChartWindow)
+          (doc ("Construct a new " ,ChartWindow " with a new width. "
+                "By default, the width will be 800")))
+        (method-spec
+          (name "height")
+          (arity 2)
+          (params ())
+          (contract (a-arrow ,Self ,N ,ChartWindow))
+          (args ("self" "height"))
+          (return ,ChartWindow)
+          (doc ("Construct a new " ,ChartWindow " with a new height. "
+                "By default, the height will 600")))
+        (method-spec
+          (name "display")
+          (arity 1)
+          (params ())
+          (contract (a-arrow ,Self ,Image))
+          (args ("self"))
+          (return ,Image)
+          (doc ("Display the chart on an interactive dialog, "
+                "and produce an " ,Image " after the dialog is closed.")))
+        (method-spec
+          (name "get-image")
+          (arity 1)
+          (params ())
+          (contract (a-arrow ,Self ,Image))
+          (args ("self"))
+          (return ,Image)
+          (doc ("Produce an " ,Image " of the chart"))))))
+    (constr-spec
+      (name "plot-chart-window")
+      (with-members (,x-min-meth ,x-max-meth ,y-min-meth ,y-max-meth ,x-axis-meth ,y-axis-meth
+                     ,num-samples-meth)))
+    (constr-spec
+      (name "histogram-chart-window")
+      (with-members (,x-min-meth ,x-max-meth ,y-max-meth ,x-axis-meth ,y-axis-meth)))
+    (constr-spec
+      (name "bar-chart-window")
+      (with-members (,y-min-meth ,y-max-meth ,x-axis-meth ,y-axis-meth)))
+    (constr-spec
+      (name "pie-chart-window")
+      (with-members ()))
+  ))
+
+@docmodule["chart"]{
+  The Pyret Chart library. It consists of chart, plot, and data visualization tools,
+  using @link["https://developers.google.com/chart/" "Google Charts"] as a backend.
+
+  This documentation assumes that you @pyret{include} the chart library,
+  and has imported @pyret{image-structs} as @pyret{I} (@pyret{import image-structs as I}).
+
+  @;############################################################################
+  @section{Creating a DataSeries}
+
+  Given data, before you visualize them as a chart, you need to choose what @emph{type}
+  of chart do you want. For example, you could either choose to visualize the data as
+  a bar chart or as a pie chart.
```

Comment:
"Talky" is intentional actually. I intend to mimic https://www.pyret.org/docs/latest/tables.html which is talky too. But yes, I can edit this.

Reply:
Gotcha.  In that case: I think I was reacting to an overabundance of commas in the beginning of the sentence.  It sounded in my head like a nervous student reading a speech, rather than a confident teacher explaining what to do.  Let's aim for "casual" tone, rather than "talky" -- sound better?

Reply:
Yup :)